src/biogeophys/TemperatureType.F90

```diff
@@ -1535,78 +1681,26 @@ subroutine UpdateAccVars (this, bounds)
 
        ! Accumulate and extract GDD8
-
-       ! SSR 2024-06-07: Don't wrap this do-loop in an "if it's not time to reset." Behavior would
-       ! be identical for now, but if "missed update" behavior is fixed (see ESCOMP/CTSM#2585), you
-       ! would end up updating GDD8 with uninitialized values.
-       do p = begp,endp
-          ! Avoid unnecessary calculations over inactive points
-          if (patch%active(p)) then
-             g = patch%gridcell(p)
-             if (month==1 .and. day==1 .and. secs==dtime) then
-                call markreset_accum_field('GDD8', p)  ! reset gdd
-             else if (( month > 3 .and. month < 10 .and. grc%latdeg(g) >= 0._r8) .or. &
-                  ((month > 9 .or.  month < 4) .and. grc%latdeg(g) <  0._r8)     ) then
-                rbufslp(p) = max(0._r8, min(30._r8, &
-                     this%t_ref2m_patch(p)-(SHR_CONST_TKFRZ + 8._r8))) * dtime/SHR_CONST_CDAY
-             else
-                rbufslp(p) = 0._r8      ! keeps gdd unchanged at other times (eg, through Dec in NH)
-             end if
-          end if
-       end do
-       call update_accum_field  ('GDD8', rbufslp, nstep)
-       call extract_accum_field ('GDD8', this%gdd8_patch, nstep)
+       call this%UpdateAccVars_CropGDDs(rbufslp, begp, endp, month, day, secs, dtime, nstep, 8, this%gdd8_patch, crop_inst)
 
        ! Accumulate and extract GDD10
-
-       ! SSR 2024-06-07: Don't wrap this do-loop in an "if it's not time to reset." Behavior would
-       ! be identical for now, but if "missed update" behavior is fixed (see ESCOMP/CTSM#2585), you
-       ! would end up updating GDD10 with uninitialized values.
-       do p = begp,endp
-          ! Avoid unnecessary calculations over inactive points
-          if (patch%active(p)) then
-             g = patch%gridcell(p)
-             if (month==1 .and. day==1 .and. secs==dtime) then
-                call markreset_accum_field('GDD10', p)  ! reset gdd
-             else if (( month > 3 .and. month < 10 .and. grc%latdeg(g) >= 0._r8) .or. &
-                  ((month > 9 .or.  month < 4) .and. grc%latdeg(g) <  0._r8)     ) then
-                rbufslp(p) = max(0._r8, min(30._r8, &
-                     this%t_ref2m_patch(p)-(SHR_CONST_TKFRZ + 10._r8))) * dtime/SHR_CONST_CDAY
-             else
-                rbufslp(p) = 0._r8      ! keeps gdd unchanged at other times (eg, through Dec in NH)
-             end if
-          end if
-       end do
-       call update_accum_field  ('GDD10', rbufslp, nstep)
-       call extract_accum_field ('GDD10', this%gdd10_patch, nstep)
+       call this%UpdateAccVars_CropGDDs(rbufslp, begp, endp, month, day, secs, dtime, nstep, 10, this%gdd10_patch, crop_inst)
 
        ! Accumulate and extract running 20-year means
        if (is_end_curr_year()) then
+          ! Flush, if needed
+          if (flush_gdd20 .or. this%flush_gdd20) then
```

Comment:
Why two `flush_gdd20`s?

Reply:
Cleaned up with bc055919b and d298aef94. Briefly: `this%flush_gdd20` refers to the member of `TemperatureType`, whereas `flush_gdd20` refers to the parameter from `clm_varctl`. I've now cleaned it up so that the latter is only considered during `TemperatureType` `Init()` and `Restart()`.